src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1191,10 +1187,10 @@ public function has_class( $wanted_class ): ?bool {
 			return null;
 		}
 
-		$wanted_class = strtolower( $wanted_class );
+		$wanted_class = $this->comparable_class_name( $wanted_class );
 
 		foreach ( $this->class_list() as $class_name ) {
-			if ( $class_name === $wanted_class ) {
+			if ( $this->comparable_class_name( $class_name ) === $wanted_class ) {
```

Comment:
in a case like this it seems like it would be better to extract the same logical condition check outside the `foreach` loop rather than repeat it on every iteration. we know that the quirks mode won't change during the loop, so we could create two loops - one for each mode.

Reply:
casing here is the only consequence and "is quirks mode" is the only check, right? so could we not inline this with `substr_compare()` and avoid the function call?

```php
$wanted_length    = strlen( $wanted_class );
$case_insensitive = self::QUIRKS_MODE === $this->compatibility_mode;

foreach ( $this->class_list() as $class_name ) {
	if (
		strlen( $class_name ) === $wanted_length &&
		0 === substr_compare( $class_name, $wanted_class, 0, $wanted_length, $case_insensitive )
	) {
		return true;
	}
```

Reply:
We can do that, but if take that approach it may be best to inline the `comparable_class_name` logic everywhere so that it's not assumed that function is not applied everywhere it's needed.

I'll push those changes and we can decide if that's preferable.

Reply:
I've pushed 15d479e and 8dbf6ab to remove the `comparable_class_name` method.